enigma/src/main/java/cuchaz/enigma/translation/representation/TypeDescriptor.java

```diff
@@ -43,7 +42,7 @@ public TypeDescriptor(String desc) {
 
 	public static String parseFirst(String in) {
 
-		if (in == null || in.length() <= 0) {
+		if (in == null || in.length() == 0) {
```

Comment:
Have you verified this and other locations don't carry the risk of being less than 0?

Reply:
Have you ever seen a string that is -1 codepoints long? It's fair to assume that in the event of there being an integer overflow that we have larger worries (if we were to assume that the JVM supports arrays of that size - which I don't think it does)